whitepaper/whitepaper.md

```diff
@@ -124,13 +134,235 @@ Blockchain technologies are growing in usage, but fragmentation is a big problem
+
+## 2.1 Terminology of Blockchain Interoperability
+
+
+This section acts as a building block to describe the different flavors of blockchain interoperability. Business use cases will be built on these simple foundation blocks leveraging a mix of them simultaneously and even expanding to several blockchains interacting concurrently. 
+
+
+### 2.1.1 Ledger Object Types
+
+
+To describe typical interoperability patterns between different blockchains, the types of objects stored on a ledger have to be distinguished. The following three types of objects stored on a ledger are differentiated as follows:
+
+* FA: Fungible asset (value token/coin)  – cannot be duplicated on different ledgers
+* NFA: Non-fungible asset – cannot be duplicated on different ledgers<sup>[d](#22-footnotes-introduction)</sup>
+* D: Data – can be duplicated on different ledgers
+
+
+Difference between a fungible (FA) and a non-fungible asset (NFA)
+
+A fungible asset is an asset that can be used interchangeably with another asset of the same type, like a currency. For example, a 1 USD bill can be swapped for any other 1 USD bill. Cryptocurrencies, such as ETH (Ether) and BTC (Bitcoin), are FAs. A non-fungible asset is an asset that cannot be swapped as it is unique and has specific properties. For example, a car is a non-fungible asset as it has unique properties, such as color and price. CryptoKitties are NFAs as well. There are two standards for fungible and non-fungible assets on the Ethereum network (ERC-20 Fungible Token Standard and ERC-721 Non-Fungible Token Standard). 
+
+Difference between an asset (FA or NFA) and data (D)
+
+Unicity applies to FAs and NFAs meaning it guarantees that only one valid representation of a given asset exists in the system. It prevents double-spending of the same token/coin in different blockchains. The same data package can have several representations on different ledgers while an asset (FA or NFA) can have only one representation active at any time, i.e., an asset exists only on one blockchain while it is locked/burned on all other blockchains. If fundamental disagreement persists in the community about the purpose or operational upgrades of a blockchain, a hard fork can split a blockchain creating two representations of the same asset to coexist. For example, Bitcoin split into Bitcoin and Bitcoin Cash in 2017. Forks are not addressing blockchain interoperability so the definition of unicity applies in a blockchain interoperability context. A data package that was once created as a copy of another data package might divert from its original one over time because different blockchains might execute different state changes on their data packages. 
+
+
+### 2.1.2 Blockchain Interoperability Types
```

Comment:
Please check page 18 from https://arxiv.org/pdf/2005.14282.pdf, and see if that classification makes sense for you.

This classification is not very clear, as the different types overlap. For instance, an atomic swap is a ledger transfer, as it locks and releases tokens, but it is also a ledger interaction, because there is a pair of actions issued in different ledgers.

Reply:
The main difference between atomic swaps and ledger transfers are the number of assets in each blockchain after the exchange. If there are 100 tokens in Blockchain A and 100 tokens in Blockchain B initially then after an atomic swap the number is still the same in each blockchain, but after a ledger transfer of one token, there are 99 tokens in Blockchain A and 101 tokens in blockchain B.
I am aware that “ledger interaction” is probably a quite broad definition and ledger transfers can be classified as a very special case of ledger interactions. Nonetheless, because ledger transfers and atomic swaps have been already quite widely used in cryptocurrency blockchains, they should be described as separate types in my opinion. To not bore the reader with any further sub-classification, I kept the definition of ledger interaction quite broad so it can include all the blockchain interoperability cases that are not ledger transfers/atomic swaps.

Reply:
Alright, understood. The mentioned classification refers to those as well, and at a later point we may discuss if we want to adopt it. In my point of view, it helps a lot classifying the existing solutions in a very clear way.